sources/packages/api/src/services/application/application.service.ts

```diff
@@ -527,11 +535,67 @@ export class ApplicationService extends RecordDataModelService<Application> {
 
   /**
    * get all student applications.
+   * @param page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @param pageLimit, limit of the page if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @param sortField, field to be sorted
+   * @param sortOrder, order to be sorted
    * @param studentId student id .
    * @returns student Application list.
    */
-  async getAllStudentApplications(studentId: number): Promise<Application[]> {
-    return this.repo
+  async getStudentAppicationAndProcessDTO(
+    sortField: string,
+    studentId: number,
+    page = DEFAULT_PAGE_NUMBER,
+    pageLimit = DEFAULT_PAGE_LIMIT,
+    sortOrder = FieldSortOrder.ASC,
+  ): Promise<StudentApplicationAndCount> {
+    const applicationsAndCount = await this.getAllStudentApplications(
+      sortField,
+      studentId,
+      page,
+      pageLimit,
+      sortOrder,
+    );
+
+    return {
+      applications: applicationsAndCount[0].map((application: Application) => {
```

Comment:
why are we doing the DTO transformation in service? we can do the service model in between to pass the paginated object.